app/scripts/metamask-controller.js

```diff
@@ -1253,6 +1253,7 @@ module.exports = class MetamaskController extends EventEmitter {
     engine.push(createOriginMiddleware({ origin }))
     engine.push(createLoggerMiddleware({ origin }))
     engine.push(filterMiddleware)
+    engine.push(require('./lib/metasea')) // hardcode plugin addition for now
```

Comment:
Very clean mvp strategy, this is the file I should've started with. What this tells me right away is that you aren't leveraging MetaMask's stored secrets. MetaMask *could* be leveraged to generate the secret material used for encryption/decryption. Eventually, this would also include the interface to forward these methods to a hardware wallet.

I'm still curious what methods MetaMask's key management would need to expose to SEA to allow it to leverage the user's existing key management strategies.

Reply:
I'd love for this to happen, the problem is knowing how to do that is way way way way outside my MetaMask understanding. Would need somebody on your team to contrib the code - I'll happily make time to answer any Qs they have.

Most of it is just following the same repeating stub/RPC pattern as https://github.com/MetaMask/metamask-extension/pull/5024/files#diff-bdf2cd624ece08f218a73a9d22e80f10R144 and just replacing the interior function call with `whatever_your_crypto_library_is` and using Aaron's RPC middleware to detect & prioritize it, like I tried to follow. Nothing here specific to my code.

Ideally somebody could do this within next 3 months, but knowing how busy you guys are I'm realistically guessing 7 months - it will be easy to evolve/upgrade into it without breaking anything.